tests/conftest.py

```diff
@@ -9,13 +9,16 @@
 
 
 @pytest.fixture(autouse=True)
-def reset_cuda_device_after_test():
+def reset_cuda_device_after_test(request):
     """Calls cudaDeviceReset() after each test.
 
     This is needed for 'cuda-memcheck --leak-check full' to catch leaks"""
 
     yield
 
+    # If the test is not marked for memory tests, no need to reset device
+    if request.node.get_closest_marker("memcheck") is None:
```

Comment:
This saved about 30 seconds for `make unit_tests` locally

Reply:
nit: [this method](https://stackoverflow.com/a/38763328/1427284) looks a little cleaner to my eye (assuming it's actually equivalent)